planner/core/planbuilder.go

```diff
@@ -1607,44 +1608,97 @@ func (b *PlanBuilder) buildLoadStats(ld *ast.LoadStatsStmt) Plan {
 	return p
 }
 
-func (b *PlanBuilder) buildSplitIndexRegion(node *ast.SplitIndexRegionStmt) (Plan, error) {
+const maxSplitRegionNum = 1000
+
+func (b *PlanBuilder) buildSplitRegion(node *ast.SplitRegionStmt) (Plan, error) {
 	tblInfo := node.Table.TableInfo
-	indexInfo := tblInfo.FindIndexByName(strings.ToLower(node.IndexName))
+	indexInfo := tblInfo.FindIndexByName(node.IndexName.L)
 	if indexInfo == nil {
 		return nil, ErrKeyDoesNotExist.GenWithStackByArgs(node.IndexName, tblInfo.Name)
 	}
+	mockTablePlan := LogicalTableDual{}.Init(b.ctx)
+	schema := expression.TableInfo2SchemaWithDBName(b.ctx, node.Table.Schema, tblInfo)
+	mockTablePlan.SetSchema(schema)
 
-	indexValues := make([][]types.Datum, 0, len(node.ValueLists))
-	for i, valuesItem := range node.ValueLists {
-		if len(valuesItem) > len(indexInfo.Columns) {
-			return nil, ErrWrongValueCountOnRow.GenWithStackByArgs(i + 1)
-		}
-		valueList := make([]types.Datum, 0, len(valuesItem))
+	checkValue := func(valuesItem []ast.ExprNode) ([]types.Datum, error) {
+		values := make([]types.Datum, 0, len(valuesItem))
 		for j, valueItem := range valuesItem {
-			x, ok := valueItem.(*driver.ValueExpr)
+			var expr expression.Expression
+			var err error
+			switch x := valueItem.(type) {
+			case *driver.ValueExpr:
+				expr = &expression.Constant{
+					Value:   x.Datum,
+					RetType: &x.Type,
+				}
+			default:
+				expr, _, err = b.rewrite(valueItem, mockTablePlan, nil, true)
+				if err != nil {
+					return nil, err
+				}
+			}
+			constant, ok := expr.(*expression.Constant)
 			if !ok {
 				return nil, errors.New("expect constant values")
 			}
+			value, err := constant.Eval(chunk.Row{})
+			if err != nil {
+				return nil, err
+			}
 			colOffset := indexInfo.Columns[j].Offset
-			value, err := x.Datum.ConvertTo(b.ctx.GetSessionVars().StmtCtx, &tblInfo.Columns[colOffset].FieldType)
+			value, err = value.ConvertTo(b.ctx.GetSessionVars().StmtCtx, &tblInfo.Columns[colOffset].FieldType)
 			if err != nil {
 				return nil, err
 			}
+			values = append(values, value)
+		}
+		return values, nil
+	}
+	p := &SplitRegion{
+		TableInfo: tblInfo,
+		IndexInfo: indexInfo,
+	}
+	if len(node.SplitOpt.ValueLists) > 0 {
+		indexValues := make([][]types.Datum, 0, len(node.SplitOpt.ValueLists))
+		for i, valuesItem := range node.SplitOpt.ValueLists {
+			if len(valuesItem) > len(indexInfo.Columns) {
+				return nil, ErrWrongValueCountOnRow.GenWithStackByArgs(i + 1)
+			}
+			values, err := checkValue(valuesItem)
+			if err != nil {
+				return nil, err
+			}
+			indexValues = append(indexValues, values)
+		}
+		p.ValueLists = indexValues
+		return p, nil
+	}
 
-			valueList = append(valueList, value)
+	checkMinMaxValue := func(valuesItem []ast.ExprNode, name string) ([]types.Datum, error) {
```

Comment:
Please use two functions to handle (min,max) syntax and valuesList syntax severally. To make your code cleaner.

Reply:
I will add some comment for this.